packages/main/src/RPA/MSGraph.py

```diff
@@ -0,0 +1,956 @@
+                    Log    ${file.created}
+                    Log    ${file.is_file}
+                    Log    ${file.is_image}
+                    Log    ${file.is_photo}
+                    Log    ${file.is_folder}
+                    Log    ${file.size}
+                    Log    ${file.web_url}
+                END
+        """
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        folder = self._get_folder_instance(drive_instance, target_folder)
+        items = folder.get_items()
+        if include_folders:
+            return items
+        return [item for item in items if not item.is_folder]
+
+    @keyword
+    def download_file_from_onedrive(
+        self,
+        target_file: Union[drive.File, str],
+        to_path: Union[Path, str, None] = None,
+        name: Optional[str] = None,
+        resource: Optional[str] = None,
+        drive_id: Optional[str] = None,
+    ) -> Path:
+        """Downloads a file from OneDrive.
+
+        The downloaded file will be saved to a local path.
+
+        :param target_file: `DriveItem` or file path of the desired file.
+        :param to_path: Destination folder of the downloaded file,
+                defaults to the current directory.
+        :param name: New name for the downloaded file, with or without extension.
+        :param resource: Name of the resource if not using default.
+        :param drive_id: Drive ID if not using default.
+        :return: Path to the downloaded file.
+
+        .. code-block: robotframework
+
+            *** Tasks ***
+            Download file with path
+                ${download_path}=    Download File From Onedrive
+                ...    /path/to/onedrive/file
+                ...    /path/to/local/folder
+                ...    Report.pdf
+
+            Download file with object
+                ${download_path}=    Download File From Onedrive
+                ...    ${drive_item}
+                ...    /path/to/local/folder
+                ...    Report.pdf
+        """
+        self._require_authentication()
+        drive_instance = self._get_drive_instance(resource, drive_id)
+        file_instance = self._get_file_instance(target_file, drive_instance)
+        return self._download_file(file_instance, to_path, name)
+
+    @keyword
+    def download_folder_from_onedrive(
+        self,
+        target_folder: Union[drive.Folder, str],
+        to_path: Union[Path, str, None] = None,
+        resource: Optional[str] = None,
+        drive_id: Optional[str] = None,
+    ) -> Path:
+        """Downloads a folder from OneDrive with all of it's contents,
+        including subfolders.
+
+        Caution when downloading big folder structures. The downloaded
+        folder will be saved to a local path.
+
+        :param target_folder: `DriveItem` or path of the desired folder.
+        :param to_path: Destination folder where the download will be saved to,
```

Comment:
possible formatting problem (I think one space is enough for the continuation line)